pkg/auth/permissions/resource.go

```diff
@@ -43,6 +45,11 @@ func (m ResourceMetadata) GetResource() Resource {
 
 // GetScope returns the resource scope for this metadata object.
 func (m ResourceMetadata) GetScope() ResourceScope {
+	// Replacing resources _may_ have a different ResourceScope than the initial resource.
+	// This way, clients don't have to deal with replacing resources when checking for scope.
+	if m.ReplacingResource != nil && m.ReplacingResource.GetScope() < m.Scope {
```

Comment:
I hung here for a while. What this actually does is _broadens_ the scope, i.e., from namespace to cluster if the replacing resource has the broader one. If so, could you please adjust the comment to make this clear?

Reply:
I've made the changes within #1834 , let's move the discussion there, if there are any additional comments.